boards nordic: Replace nrf54l20pdk board with nrf54lm20pdk

aescolar opened the pull request:
Replace the nRF54L20PDK board and nRF54L20 SOC definitions with the nRF54LM20 & nRF54LM20PDK ones from NCS together with the necessary changes to the SOC, DTS and nordic hal integration to support it. As well as replacing all tests/samples overlays and filters from one to the other.

aescolar committed:
boards / soc nordic: Add nrf54lm20A SOC & nrf54lm20pdk board

Bring the nRF54LM20A SOC and nRF54LM20PDK board definitions from NCS
together with the necessary changes to the SOC, DTS and nordic hal
integration to support it.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

carlescufi reviewed (changes requested):
Please remove the existing `nrf54l20pdk` at the same time

aescolar committed:
boards/nordic/nrf54lm20pdk: Take docs from nrf54l20pdk

And change all mentions of the old board/SOC name to the new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/uart: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old name
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/spi: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old name
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/gpio: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old name
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/flash: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/clock_control: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/counter: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/drivers/mbox: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/watchdog: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/i2c: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/retained_mem: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/adc: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/boards/nrf/qdec: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/boards/nrf/i2c: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/boards/nrf/hwinfo: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/timer/nrf_grtc_timer: Switch nrf54l20pdk w nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/boards/nordic/system_off: Switch nrf54l20pdk w nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/subsys/fs/fcb: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/i2s: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/drivers/audio: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/drivers/watchdog: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/audio: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/sensor: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/drivers/counter: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/kernel/timer: Switch nrf54l20pdk with nrf54lm20pdk

Switch in filters, and overlays the board name, as the old one
is being removed and replaced by this new one.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
boards/nordic: Remove nrf54l20pdk

It has been replaced by the nrf54lm20pdk now

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
soc / DTS: Remove nRF54L20

Remove the SOC & DTS definitions, kconfig options and nordic
hal adaptation for this SOC.
The SOC was renamed nRF54LM20, and as such were replaced
the corresponding files in Zephyr.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
release notes: Remove link to board which is going to be removed

The nrf54L20 board (and its documentation) are going to be removed.
Let's remove the link so the documentation can still be built.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
